tests/workflows/basic_tool_calling_node/tests/test_workflow.py

```diff
@@ -0,0 +1,256 @@
+from vellum.client.types.prompt_request_string_input import PromptRequestStringInput
+from vellum.client.types.rich_text_prompt_block import RichTextPromptBlock
+from vellum.client.types.string_vellum_value import StringVellumValue
+from vellum.client.types.variable_prompt_block import VariablePromptBlock
+from vellum.client.types.vellum_variable import VellumVariable
+from vellum.workflows.nodes.displayable.bases.inline_prompt_node.constants import DEFAULT_PROMPT_PARAMETERS
+
+from tests.workflows.basic_tool_calling_node.workflow import GetCurrentWeatherWorkflow
+
+
+def test_get_current_weather_workflow(vellum_adhoc_prompt_client, mock_uuid4_generator):
+    """
+    Test that the GetCurrentWeatherWorkflow returns the expected outputs.
+    """
+
+    def generate_prompt_events(*args, **kwargs) -> Iterator[ExecutePromptEvent]:
+        execution_id = str(uuid4())
+
+        call_count = vellum_adhoc_prompt_client.adhoc_execute_prompt_stream.call_count
+        expected_outputs: List[Union[FunctionCallVellumValue, StringVellumValue]]
+        if call_count == 1:
+            expected_outputs = [
+                FunctionCallVellumValue(
+                    value=FunctionCall(
+                        arguments={"location": "Miami", "unit": "metric"},
+                        id="call_7115tNTmEACTsQRGwKpJipJK",
+                        name="get_current_weather",
+                        state="FULFILLED",
+                    ),
+                ),
+            ]
+        else:
+            expected_outputs = [StringVellumValue(value="The current temperature in San Francisco is 22°C (71.6°F).")]
+
+        events: List[ExecutePromptEvent] = [
+            InitiatedExecutePromptEvent(execution_id=execution_id),
+            FulfilledExecutePromptEvent(
+                execution_id=execution_id,
+                outputs=expected_outputs,
+            ),
+        ]
+        yield from events
+
+    # Set up the mock to return our events
+    vellum_adhoc_prompt_client.adhoc_execute_prompt_stream.side_effect = generate_prompt_events
+
+    uuid4_generator = mock_uuid4_generator("vellum.workflows.nodes.displayable.bases.inline_prompt_node.node.uuid4")
+    first_call_input_id = uuid4_generator()
+    second_call_input_id = uuid4_generator()
+
+    # GIVEN a get current weather workflow
+    workflow = GetCurrentWeatherWorkflow()
+
+    # WHEN the workflow is executed
+    terminal_event = workflow.run()
+
+    # THEN the workflow is executed successfully
+    assert terminal_event.name == "workflow.execution.fulfilled"
+    assert terminal_event.outputs.text == "The current temperature in San Francisco is 22°C (71.6°F)."
+    assert terminal_event.outputs.chat_history == [
+        ChatMessage(
+            text='{\n    "arguments": {\n        "location": "Miami",\n        "unit": "metric"\n    },\n    "id": "call_7115tNTmEACTsQRGwKpJipJK",\n    "name": "get_current_weather",\n    "state": "FULFILLED"\n}',  # noqa: E501
```

Comment:
[nit] I would expect this ChatMessage to contain a `FUNCTION_CALL` content rather than text